sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/runners/inprocess/InProcessCreate.java

```diff
@@ -204,6 +209,32 @@ public T getCurrent() throws NoSuchElementException {
 
       @Override
       public void close() throws IOException {}
+
+      @Override
+      public Double getFractionConsumed() {
+        return rangeTracker.getFractionConsumed();
+      }
+
+      @Override
+      public InMemorySource<T> splitAtFraction(double fraction) {
+        long splitPosition = rangeTracker.getPositionForFractionConsumed(fraction);
+        if (rangeTracker.trySplitAtPosition(splitPosition)) {
+          List<byte[]> oldSourceElems =
+              ImmutableList.copyOf(source.allElementsBytes.subList(0, (int) splitPosition));
+          List<byte[]> newSourceElems =
+              ImmutableList.copyOf(
+                  source.allElementsBytes.subList(
```

Comment:
I've removed some construction which wasn't useful, changed the conditionals, and added a bit of symmetry in how the list is split. I could move the splitting into the source proper, but don't think it's a particularly large improvement.